wasm/src/lib.rs

```diff
@@ -397,6 +412,32 @@ pub fn run(
         // See example at https://rustwasm.github.io/wasm-bindgen/reference/receiving-js-closures-in-rust.html
         let _ = event_cb.call1(&JsValue::null(), &JsValue::from(msg));
     };
+
+    // See if the pauliNoise JsValue is an array
+    let noise = if pauliNoise.is_array() {
+        let pauliArray = js_sys::Array::from(pauliNoise);
+        if pauliArray.length() != 3 {
+            return Err(JsError::new("Pauli noise must have 3 probabilities").into());
+        }
+        PauliNoise::from_probabilities(
+            pauliArray
+                .get(0)
+                .as_f64()
+                .expect("Probabilities should be floats"),
+            pauliArray
+                .get(1)
+                .as_f64()
+                .expect("Probabilities should be floats"),
+            pauliArray
+                .get(2)
+                .as_f64()
+                .expect("Probabilities should be floats"),
+        )
+        .expect("Unable to create Pauli noise from the array provided")
```

Comment:
Will this expect result in panic? We probably want to do the return 'Err(JsError::new( ... ))' and return the error string from underlying Err case. These numbers come from a user input and shouldn't result in Panic.
Even if you are checking all conditions for Pauli noise in your code...

Reply:
It should return an error. (Also, only VS Code calls this currently and has already done a bunch of validation, so should never happen).